Support device names

NicoWeio opened the pull request:
Apart from translations, this should be ready for a merge. Hope you like it! :)

thorpelawrence commented:
Thanks for this! Were you able to get the tests to pass? For some reason it looks like travis has not picked up this PR

NicoWeio commented:
Maybe you've disabled CI for PRs? 
Anyway, my code actually wasn't that ready for a merge: I fixed two bugs, and fixed the testing. As I modified some strings, there are now missing translations in es, fr, it and pt.

thorpelawrence commented:
Looks like travis is just being slow today, and it's passed now so thanks - as you say about the translation, I'm not sure what to do about that as they were contributed by other people, so maybe they will need to be fixed later
